src/stan/io/deserializer.hpp

```diff
@@ -0,0 +1,1978 @@
+      ret.emplace_back(this->read_lub<value_type_t<Ret>>(lb, ub[i], sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Specialization for `lub` constrain with an `std::vector` return type
+   * with `std::vector` lower bound and upper bound.
+   *
+   * @tparam Ret The type to return.
+   * @tparam LB Type of lower bound.
+   * @tparam UB Type of upper bound.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param lb Lower bound.
+   * @param ub Upper bound.
+   * @param vecsize The size of the return `std::vector`
+   * @param sizes The dimensions for the inner type.
+   * @throw std::runtime_error if the scalar is not between the specified
+   *    lower and upper bounds.
+   */
+  template <typename Ret, typename LB, typename UB, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  auto read_lub(const std::vector<LB>& lb, const std::vector<UB>& ub,
+                size_t vecsize, Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(
+          this->read_lub<value_type_t<Ret>>(lb[i], ub[i], sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Return the next object transformed to be between the
+   * the specified lower and upper bounds.
+   *
+   * <p>See <code>stan::math::lub_constrain(T, double, double, T&)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam LB Type of lower bound.
+   * @tparam UB Type of upper bound.
+   * @tparam LP Type of log probability.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param lb Lower bound.
+   * @param ub Upper bound.
+   * @param lp Reference to log probability variable to increment.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   */
+  template <typename Ret, bool Jacobian, typename LB, typename UB, typename LP,
+            typename... Sizes>
+  auto read_lub(const LB& lb, const UB& ub, LP& lp, Sizes... sizes) {
+    if (Jacobian) {
+      return stan::math::lub_constrain(this->read<Ret>(sizes...), lb, ub, lp);
+    } else {
+      return stan::math::lub_constrain(this->read<Ret>(sizes...), lb, ub);
+    }
+  }
+
+  /**
+   * Return the next object, checking that it's elements are positive.
+   *
+   * <p>See <code>stan::math::check_positive(T)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   *
+   * @throw std::runtime_error if x is not positive
+   */
+  template <typename Ret, typename... Sizes>
+  auto read_pos(Sizes... sizes) {
+    auto ret = read<Ret>(sizes...);
+    using stan::math::check_positive;
+    using stan::math::value_of;
+    check_positive("deserializer", "Positive Constrained", value_of(ret));
+    return ret;
+  }
```

Comment:
I don't see an explicit positive constrained variable in the Stan reference manual. Maybe this is a leftover type from earlier that we don't need to re-implement?

Reply:
Oh? I mean maybe this is an easy type to add to the language? It seems reasonable and we have all the stuff in math for it

Reply:
I'm gonna remove this. It's just easy enough to say in the language:

```
real<lower = 0.0> x;
```

Reply:
Lol oh yeah true true